k8s cluster monitoring (#79)
broadinstitute/terraform-shared: terraform-modules/stackdriver/k8s-cluster-monitoring/README.md

### K8s-node-health
This policy monitors a variety of different indicators of node health using passing/failing states. These indicators are: 
- Corrupt Docker Overlay
- Disk Pressure
- Memory Pressure
- Frequent Container Restarts
- Frequent Docker Restarts
- Frequent Kubelet Restarts
- Kernel Deadlock
- Network Unavailable
- PID Pressure

If any of these are in a failing state for more than 5 minutes an alert will fire. This policy utilizes prometheus metrics

### k8s-node-resource-utilization
This policy monitors the cpu and memory utilization ratios of each node in a cluster. This alert policy will trigger if either resource utilization is above a set threshold percentage for greater than 5 minutes. Not reliant on prometheus

### K8s-pod-health

> This policy monitors for any pods that are on able to be run by the scheduler for an extended period of time. This alert fires if a pod is in failed to schedule state for greater than 5 minutes. The alert will provide which cluster and namespace are associated with the pod that triggered the alert.

Reviewer: ```suggestion
This policy monitors for any pods that are not able to be run by the scheduler for an extended period of time. This alert fires if a pod is in failed to schedule state for greater than 5 minutes. The alert will specify which cluster and namespace are associated with the pod that triggered the alert.
```

> ### K8s-pod-resource utilization 

Reviewer 2: Make sure this can be turned off for certain pods.  Sometimes services with memory leaks need to be deployed, and then they use the kubernetes OOM kill process to flush it with a pod restart as part of a normal operational process.

Author: Hmm I know you can definitely apply filters, by k8s resources such as pod name or deployment name. However these filters are baked into the metric strings that are found in the locals section of each file. Would have to do some thinking on how to make this toggleable for specific pods. Perhaps an optional variable that adds filters to the end of the string if specified

Author: THe easiest way might be to have any service that does this tag its pods with a specific label, then we could filter for that label

This policy monitors the ratio of the current cpu and memory being used by a pod to the limit for that resource set in the pod spec. It also monitors volume utilization at the pod level. This is important to monitor because it can cause the scheduler to OOMkill pods automatically potentially killing crucial application pods. Not reliant on prometheus. THe alert will provide the cluster and namespace of the pod that triggered the alert.

## Notifications

> Currently the default on this module is to send all alert notifications to ap-devops slack. A potential feature to add would to have notification channels configured based on a passed in environment ie. pager duty for prod, slack for others.

Reviewer 2: This is a great feature.
